pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -137,6 +137,39 @@ func (r Results) NonPendingPodSchedulingErrors() string {
 	return msg.String()
 }
 
+func (r Results) NodeClaimsMeetingMinimumRequirements() Results {
```

Comment:
```suggestion
func (r Results) TruncateInstanceTypes(int maxInstanceTypes) Results {
```

nit: It's a tad odd to have a name here that indicates that you are just returning NodeClaims from reading it but actually returns an entire results object. Consider allowing the truncation point be configurable to generalize it.